packages/widget/src/hooks/useTxHistory.ts

```diff
@@ -1,35 +1,63 @@
 import { TxsStatus, useBroadcastedTxsStatus } from "@/pages/SwapExecutionPage/useBroadcastedTxs";
+import { useSyncTxStatus } from "@/pages/SwapExecutionPage/useSyncTxStatus";
+import { removeTransactionHistoryItemAtom, TransactionHistoryItem } from "@/state/history";
+import { skipChainsAtom } from "@/state/skipClient";
 import { SimpleStatus } from "@/utils/clientType";
-import { TransferAssetRelease } from "@skip-go/client";
 import { useQuery } from "@tanstack/react-query";
+import { useAtomValue, useSetAtom } from "jotai";
 
-export const useTxHistory = ({
-  txs,
-  txsRequired,
-}: {
-  statusData?: TxsStatus;
-  txs: {
-    chainID: string;
-    txHash: string;
-  }[];
-  txsRequired: number;
-}): {
-  status?: SimpleStatus;
-  transferAssetRelease?: TransferAssetRelease;
-  explorerLinks: string[];
-} => {
-  const {
-    data: statusData,
-    isFetching,
-    isPending,
-  } = useBroadcastedTxsStatus({
-    txsRequired: txsRequired,
+type useTxHistoryProps = {
+  index: number;
+  txHistoryItem?: TransactionHistoryItem;
+};
+
+export const useTxHistory = ({ txHistoryItem, index }: useTxHistoryProps) => {
+  const { data: chains } = useAtomValue(skipChainsAtom);
+
+  const removeTransactionHistoryItem = useSetAtom(removeTransactionHistoryItemAtom);
+
+  const txs = txHistoryItem?.transactionDetails?.map((tx) => ({
+    chainID: tx.chainID,
+    txHash: tx.txHash,
+  }));
+
+  const txHashFoundInTxHistoryItem = txs?.some((tx) => tx.txHash);
+
+  const chainIdFound = chains?.some((chain) =>
+    txs?.map((tx) => tx.chainID).includes(chain.chainID),
+  );
+
+  const txsRequired = txHistoryItem?.route?.txsRequired;
+
+  const shouldFetchStatus =
+    txHistoryItem?.lastTxStatus !== "failed" &&
+    txHistoryItem?.lastTxStatus !== "success" &&
+    txs !== undefined &&
+    chainIdFound;
```

Comment:
we cant rely on `isSettled`, because it's possible that `isSettled` will never resolve, if the second tx doesn't exist / was never added to the txHistoryItem

Reply:
or actually maybe i can add sometype of limit of number of req that we do to useQuery

Reply:
with 2tx routes i think this might be problematic

Reply:
maybe i can just check txs.length and txsRequired